### PR TITLE
Fix Warnings in Vala Code

### PR DESCRIPTION
**Pull request by marbetschar**
This PR fixes all the warnings in the Vala codebase - making it a bit easier to work on new stuff.

### CODE REVIEW

**Comment by Marukesu** [comment]
looks like there's still one:
```
../src/FoldersView/GroupedFolderSourceItem.vala:157.13-157.57: warning: switch does not handle `FLAGS_LAST', `NOCHILDREN', `TYPE_NORMAL', `TYPE_TASKS', `SYSTEM', `TYPE_TRASH', `TYPE_DRAFTS', `TYPE_CONTACTS', `NOINFERIORS', `TYPE_EVENTS', `SUBSCRIBED', `TYPE_INBOX', `TYPE_ALL', `NOSELECT', `VTRASH', `TYPE_JUNK', `READONLY', `FLAGGED', `CHILDREN', `SHARED_BY_ME', `TYPE_MEMOS', `VIRTUAL', `TYPE_OUTBOX', `TYPE_ARCHIVE', `SHARED_TO_ME', `WRITEONLY' of enum `Camel.FolderInfoFlags'
            switch (folder_type & Camel.FOLDER_TYPE_MASK) {
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by marbetschar** [comment]
Interesting, this warning was not displayed on my end for whatever reason. Anyway, its fixed now too.

**Comment by danrabbit** [comment]
@marbetschar can you resolve conflicts in this branch? :)

**Comment by marbetschar** [comment]
@danrabbit done.

File: src/FoldersView/GroupedFolderSourceItem.vala [hunk 1]
```diff
@@ -154,13 +154,13 @@ public class Mail.GroupedFolderSourceItem : Granite.Widgets.SourceList.Item {
...
-                        return Utils.strip_folder_full_name (account.service.uid, mail_submission_extension.dup_sent_folder ());
-                    }
-                    break;
+            if (
+                Camel.FolderInfoFlags.TYPE_SENT == (folder_type & Camel.FOLDER_TYPE_MASK)
+                &&
+                identity_source.has_extension (E.SOURCE_EXTENSION_MAIL_SUBMISSION)
+            ) {
+                var mail_submission_extension = (E.SourceMailSubmission) identity_source.get_extension (E.SOURCE_EXTENSION_MAIL_SUBMISSION);
+                return Utils.strip_folder_full_name (account.service.uid, mail_submission_extension.dup_sent_folder ());
```
Comment (tintou) [inline src/FoldersView/GroupedFolderSourceItem.vala:163]: While we are at it
```suggestion
                unowned var mail_submission_extension = (E.SourceMailSubmission) identity_source.get_extension (E.SOURCE_EXTENSION_MAIL_SUBMISSION);
                return Utils.strip_folder_full_name (account.service.uid, mail_submission_extension.sent_folder);
```

**Review by tintou — APPROVED** [review]

**Pull request merged**